server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -169,18 +169,7 @@ public RepositoryData addSnapshot(final SnapshotId snapshotId,
         newSnapshotStates.put(snapshotId.getUUID(), snapshotState);
         Map<IndexId, Set<SnapshotId>> allIndexSnapshots = new HashMap<>(indexSnapshots);
         for (final IndexId indexId : snapshottedIndices) {
-            if (allIndexSnapshots.containsKey(indexId)) {
-                Set<SnapshotId> ids = allIndexSnapshots.get(indexId);
-                if (ids == null) {
```

Comment:
Presumably at some point `allIndexSnapshots` could have contained `null`s? As far as I can tell this is no longer the case. ✅